pkg/auth/cell.go

```diff
@@ -39,46 +39,54 @@ var Cell = cell.Module(
 		// Always fail auth handler provides support for auth type "always-fail" - which always fails.
 		newAlwaysFailAuthHandler,
 	),
-	cell.Config(config{MeshAuthMonitorQueueSize: 1024}),
+	cell.Config(config{MeshAuthQueueSize: 1024}),
 	cell.Config(MTLSConfig{}),
 )
 
 type config struct {
-	MeshAuthMonitorQueueSize int
+	MeshAuthQueueSize int
 }
 
 func (r config) Flags(flags *pflag.FlagSet) {
-	flags.Int("mesh-auth-monitor-queue-size", r.MeshAuthMonitorQueueSize, "Queue size for the auth monitor")
+	flags.Int("mesh-auth-queue-size", r.MeshAuthQueueSize, "Queue size for the auth manager")
+}
+
+func newSignalRegistration(sm signal.SignalManager, config config) (<-chan AuthKey, error) {
+	var signalChannel = make(chan AuthKey, config.MeshAuthQueueSize)
+
+	// RegisterHandler registers signalChannel with SignalManager, but flow of events
+	// starts later during the OnStart hook of the SignalManager
+	err := sm.RegisterHandler(signal.ChannelHandler(signalChannel), signal.SignalAuthRequired)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to set up signal channel for datapath authentication required events: %w", err)
+	}
+	return signalChannel, nil
 }
 
 type authManagerParams struct {
 	cell.In
 
 	Lifecycle             hive.Lifecycle
-	Config                config
+	SignalChannel         <-chan AuthKey
```

Comment:
wasn't aware that this works with channels. might be a little bit problematic if someone else is providing a channel with the same type. but would result in an error - and therefore fine for the moment :+1: